x-pack/plugins/index_management/public/application/hooks/use_index_functions.ts

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { useCallback, useState } from 'react';
+import { RouteComponentProps } from 'react-router-dom';
+import { i18n } from '@kbn/i18n';
+import { Index } from '../../../common';
+import { Section } from '../../../common/constants';
+import { resetIndexUrlParams } from '../sections/home/index_list/details_page/reset_index_url_params';
+import { deleteIndices, loadIndex } from '../services';
+import { Error } from '../../shared_imports';
+import { notificationService } from '../services/notification';
+
+export const useIndexFunctions = (
```

Comment:
nit: I would split this hook in two or even more to follow single responsibility principle